### ISSUE TITLE
Is there a way to configure a client?

### ISSUE THREAD

**Issue by tarjei** [luxflux/puppet-openvpn#121]
Hi, I'm a bit confused as to how I setup a client using this module. If I have generated a client config, is there a way to use that config in another host using a define? If I use the openvpn::server { remote=> ...} method it doesn't seem to create the correct configuration.

kind regards,
Tarjei

**Comment by luxflux** [comment]
A configuration like this works for me:

```puppet
openvpn::server {
  'client1':
    port                => 1194,
    remote              => [ 'server1.example.org 1194', 'server2.example.org 1194' ],
    server_poll_timeout => 1,
    topology            => '',
    common_name         => 'server1-vpn',
    dev                 => 'tun0',
    compression         => '',
    persist_key         => true,
    persist_tun         => true,
    proto               => 'udp',
    verb                => 3,
}
```

What kind of errors do you see? What's wrong with the configuration?

**Comment by zipkid** [comment]
Hello,

I have the same 'confusion'.
When using the example config from the readme for a server in client mode or the one you pasted above the server.erb template is used which then has empty variables and gives the following result
```client
ns-cert-type server
remote <server> 1194
server-poll-timeout 1
ca /etc/openvpn//keys/ca.crt
cert /etc/openvpn//keys/.crt
key /etc/openvpn//keys/.key
proto udp
port 1194
group nobody
user nobody
status vpn/openvpn-status.log
dev tun0
local
push "route 192.168.192.0 255.255.255.0"
push "route 172.17.0.0 255.255.255.0"
verb 3
persist-key
persist-tun```

- The ns-cert-type is set to 'server'
- directory and key names are blank

and some other settings are not correct either.

Conditionally using the client.erb in the '# VPN Client Mode' block in server.pp does not work as then the erb parsing fails with undefined variables.

using the openvpn::client is also not an option as that seems to only 'prepare' config on the server for other clients....

Regards,

Stefan.

**Comment by luxflux** [comment]
Thanks for reporting @zipkid. I will have to look into this a bit deeper and will notify you if.

**Comment by luxflux** [comment]
I fixed this issue with #137.
The parameter `ns-cert-type` is used to validate that the certificate of the machine the client is connecting  is a `server` certificate, so this should not be an issue here.
